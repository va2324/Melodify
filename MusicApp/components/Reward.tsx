import { StyleSheet, View, Pressable, Text, Image } from 'react-native';
import { Link, Stack } from 'expo-router';
import React, { useState, useEffect } from 'react';
import { doc, getDoc, getFirestore } from 'firebase/firestore';
import { auth, db, updateCoins } from '@/firebaseConfig';
import { signInWithEmailAndPassword } from 'firebase/auth';
import {useAuth} from '../app/context/AuthContext';
import Ionicons from '@expo/vector-icons/Ionicons';

type Props = {
    reward: number;
  };
  

export default function Reward ({ reward }: Props){
  return (
    <View style= {styles.container}>
        <Ionicons name="diamond" size={24} color="#00BFFF"/>

        <Text style={styles.coins}>  :{reward}</Text>
    </View>
                      
    );
  }
  
  const styles = StyleSheet.create({
    container: { 
      borderRadius: 8,
      flexDirection: "row",
      alignItems: 'center',
      justifyContent: 'flex-end',
      width: '100%',
    },
    coins: {
      fontSize: 20,
      color: "#fff",
    }
  });
