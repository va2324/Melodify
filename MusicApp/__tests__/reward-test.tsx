import React from "react";
import renderer from 'react-test-renderer';
import Reward from "@/components/Reward";

describe('reward', () => {
    test('renders successfully', () => {
        const tree = renderer.create(<Reward reward={5}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});