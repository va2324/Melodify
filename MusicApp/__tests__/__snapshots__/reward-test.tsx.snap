// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reward renders successfully 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderRadius": 8,
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "testUri": "../../../assets/images/coin.png",
      }
    }
    style={
      {
        "height": 20,
        "width": 20,
      }
    }
  />
  <Text
    style={
      {
        "fontSize": 20,
      }
    }
  >
      :
    5
  </Text>
</View>
`;
